### DISCUSSION TITLE
Cake-Build Run Website Locally

### DISCUSSION THREAD

**Discussion by TomerPacific** [cake-build/cake#3563] [Q&A] [answered]
Hi,
I am working on a feature for the cake-build website and am trying to run things locally (on a Windows machine).
From the readme of the repository, I am supposed to run:

`build --target=Preview`

but it is not recognized in the terminal.
I assume that I need to install some form of cake and from reading the documentation, I don't understand which runner I need to download (if at all).

I would appreciate if anyone could explain to me how I can run things locally.

**Comment by pascalberger** [comment, marked as answer]
On which OS are you running? On Windows you need to run `build.ps1` on Linux/macOS `build.sh`. These bootstrapping scripts will download Cake.

**Reply by TomerPacific** [reply]
@pascalberger - I have a Windows machine using Git Bash. When I run the command you have given, I get:

```
$ ./build.ps1
./build.ps1: line 1: =: command not found
./build.ps1: line 2: =: command not found
./build.ps1: line 3: =: command not found
./build.ps1: line 4: =: command not found
./build.ps1: line 6: [string]: command not found
./build.ps1: line 7: [string]: command not found
./build.ps1: line 8: syntax error near unexpected token `$line'
./build.ps1: line 8: `foreach($line in Get-Content "$PSScriptRoot\build.config")'
```

**Reply by pascalberger** [reply]
On Windows you need to run `.\build.ps1 --target=preview` from a PowerShell session

**Comment by TomerPacific** [comment]
@pascalberger, I have overcome some intermittent issues following your latest message. Now I am getting this:
```
error NU1101: Unable to find package cake.tool. No packages exist with this id in source(s): Cake MyGet
The tool package could not be restored.
Tool 'cake.tool' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
```
When I run `nuget sources list` I get:

```
Registered Sources:
  1.  Cake MyGet [Enabled]
      https://www.myget.org/F/cake/api/v3/index.json
```

Should there be other sources I should register for?

**Reply by pascalberger** [reply]
It seems like global NuGet configuration is missing in your environment.

Can you try to replace the content of the `nuget.config` file in the root of the website repository with the following content:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="Cake MyGet" value="https://www.myget.org/F/cake/api/v3/index.json" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

**Reply by TomerPacific** [reply]
I replaced the content with what you wrote and I ran the command again. This is the result:

```
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.

dotnet-install: .NET Core SDK version 5.0.400 is already installed.
dotnet-install: Adding to current process PATH: "C:\Users\Tomer\Documents\Projects\website\.dotnet\". Note: This change will not be visible if PowerShell was run as a child process.
You can invoke the tool using the following command: dotnet-cake
Tool 'cake.tool' (version '1.2.0') was successfully installed.
A fatal error occurred. The required library hostfxr.dll could not be found.
If this is a self-contained application, that library should exist in [C:\Users\Tomer\Documents\Projects\website\tools\.store\cake.tool\1.2.0\cake.tool\1.2.0\tools\net5.0\any\].
If this is a framework-dependent application, install the runtime in the global location [C:\Program Files\dotnet] or use the DOTNET_ROOT environment variable to specify the runtime location or register the runtime location in [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x64\InstallLocation].

The .NET runtime can be found at:
  - https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=win10-x64&apphost_version=5.0.9
```

**Reply by pascalberger** [reply]
@TomerPacific can you please try again with latest version from `master` branch if you still have issues.

**Reply by TomerPacific** [reply]
When you say latest version, what are you referring to? I have forked the website project a couple days ago and I branched out from the master branch there. I have resynced my forked repository now and ran the command again. Still getting the same error as above.
What am I doing wrong?

**Reply by gep13** [reply]
Have you updated your branch with the latest changes from the upstream repository?

Typically do this by:

```
git checkout master
git fetch upstream
git rebase upstream/master
git push origin
git checkout <branch-name>
git rebase master
```

This would then mean that your fork is all updated as well as your branch having the latest changes as well.

The above won't work if you have any local pending file changes.

**Reply by TomerPacific** [reply]
@gep13 - as I have said, I resynced my fork with the latest changes from master today. 


![cake](https://user-images.githubusercontent.com/23402988/136656277-3b54d062-a53e-4843-85ea-34c29b17b866.jpg)

**Reply by augustoproiete** [reply]
Just to be sure, did you pull the latest changes on your fork to your local repo?

The latest version should be using Cake v1.3.0 and you should see a different set or messages in the log.

For instance, if you still see this message:

```
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
```

It means you don't have the latest version of the website on your machine.

**Reply by pascalberger** [reply]
It seems like Cake is not running on your environment. 

@TomerPacific Can you please run the following command in your repo and report the output:

```
dotnet tool list
```

And also the following command:

```
dotnet cake --help
```

**Reply by TomerPacific** [reply]
I did not have the .NET SDK, so I installed it. Running the command `dotnet tool list` yields:

```
PS C:\Users\Tomer\Documents\Projects\website> dotnet tool list
Welcome to .NET 5.0!
---------------------
SDK Version: 5.0.401

Telemetry
---------
The .NET tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
Package Id      Version      Commands      Manifest
```

Running the command `dotnet cake --help` yields:

```
PS C:\Users\Tomer\Documents\Projects\website> dotnet cake --help
Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET program, but dotnet-cake does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
```

**Comment by augustoproiete** [comment]
I looks like you don't have the latest version of the website on your machine... Try doing a `git pull upstream master --rebase` to get the latest version.

If you don't have the `upstream` remote defined, create it with `git remote add upstream https://github.com/cake-build/website.git`, and then do the `git pull` above

---

This is the output of `dotnet tool list` I'd expect you to see on the latest version:

```
PS C:\Users\Tomer\Documents\Projects\website> dotnet tool list
Package Id      Version      Commands         Manifest
------------------------------------------------------------------------------------------------------------------------
cake.tool       1.3.0        dotnet-cake      C:\Users\Tomer\Documents\Projects\website\.config\dotnet-tools.json
```

**Reply by TomerPacific** [reply]
@augustoproiete, pulled from master and merged it to my feature branch.
Running `build.ps1 --target=preview` seems to be working now. Thanks.

**Reply by augustoproiete** [reply]
@TomerPacific Great!

BTW, if you are using the Git Bash, you also have the option to run `./build.sh` if you prefer... We've also recently added a `build.cmd` for devs running regular Command-prompt on Windows. All the `build.*` do the same things and call `dotnet cake` so there's no difference other than convenience/preference
